### PR TITLE
Updates client library, cleans up JS, fixes package.js so the tests run

### PR DESCRIPTION
**Pull request by gguuss**

### CODE REVIEW

File: iot/manager/manager.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ function lookupRegistry (client, registryId, projectId, cloudRegion, cb) {
       console.log(err);
     } else {
       console.log('Looked up existing registry');
-      console.log(data);
+      console.log(data.data);
```
Comment (JustinBeckwith) [inline iot/manager/manager.js]: Calling this `data.data` is confusing.  The callback argument should be named `res`, or `response`, as it contains an HTTP response object.

File: iot/manager/manager.js [hunk 2]
```diff
@@ -744,23 +746,29 @@ function getRegistry (client, registryId, projectId, cloudRegion) {
...
-
-  google.discoverAPI(discoveryUrl, {}, (err, client) => {
-    if (err) {
-      console.log('Error during API discovery', err);
-      return undefined;
+  google.auth.getClient().then(authClient => {
+    const discoveryUrl =
+        `${DISCOVERY_API}?version=${API_VERSION}`;
+
+    if (authClient.createScopedRequired && authClient.createScopedRequired()) {
```
Comment (JustinBeckwith) [inline iot/manager/manager.js]: You don't need to do this :) Just pass the list of scopes into the `getClient` method:

```js
google.auth.getClient({
  scopes: ['https://www.googleapis.com/auth/cloud-platform']
}).then(auth => {
  google.options({ auth });
  google.discoverAPI(....
});
```

**Comment by gguuss** [comment]
Adds the following test fixes:

* Updates the client library to the latest (the old version I used is incompatible with GCE)
* Updates the MQTT sample to have test scripts so it actually runs tests

File: iot/mqtt_example/package.json [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
     "@google-cloud/pubsub": "0.16.4",
-    "@google-cloud/nodejs-repo-tools": "2.2.1",
+    "@google-cloud/nodejs-repo-tools": "1.4.17",
     "ava": "0.25.0",
     "jsonwebtoken": "8.2.0",
     "mqtt": "2.16.0",
     "uuid": "3.2.1",
     "yargs": "11.0.0"
   },
+  "testDependencies": {},
```
Comment (JustinBeckwith) [inline iot/mqtt_example/package.json]: I don't believe `testDependencies` are a thing :)

File: iot/mqtt_example/package.json [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
+  "name": "nodejs-docs-samples-iot-mqtt-example",
+  "scripts": {
+    "lint": "samples lint",
+    "pretest": "npm run lint",
+    "test": "samples test run --cmd ava -- -T 3m --verbose system-test/*.test.js"
+  },
   "dependencies": {
     "@google-cloud/pubsub": "0.16.4",
-    "@google-cloud/nodejs-repo-tools": "2.2.1",
+    "@google-cloud/nodejs-repo-tools": "1.4.17",
```
[the remark this answers is not in the document]
Comment (gguuss) [reply iot/mqtt_example/package.json]: samples command is in 1.x?  This affects the packages.json syntax for `test` -- couldn't find one for 2.2.1

[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply iot/mqtt_example/package.json]: @jmdobry whatup?

[the remark this answers is not in the document]
Comment (gguuss) [reply iot/mqtt_example/package.json]: I blame myself, I didn't really look very hard.

[the remark this answers is not in the document]
Comment (gguuss) [reply iot/mqtt_example/package.json]: Found it, will commit with update to runner / repo tools

[the remark this answers is not in the document]
Comment (gguuss) [reply iot/mqtt_example/package.json]: Done.

**Pull request merged**